unliftio/src/UnliftIO/IO/File.hs

```diff
@@ -89,7 +89,7 @@ import Data.ByteString as B (ByteString, writeFile)
 import Control.Monad.IO.Unlift
 import UnliftIO.IO (Handle, IOMode(..), withBinaryFile)
 
-#if WINDOWS
+#if (WINDOWS || OLD-GLIBC)
```

Comment:
It would be nice to have some kind of a changelog entry explaining this change and what the impact of enabling the flag would be.